presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotFilterExpressionConverter.java

```diff
@@ -141,6 +240,22 @@ private PinotExpression handleCast(CallExpression cast, Function<VariableReferen
             if (typeManager.canCoerce(input.getType(), expectedType)) {
                 return input.accept(this, context);
             }
+            if (expectedType == DateType.DATE) {
```

Comment:
I really liked the block comment you added on line 163: I am wondering if it is possible to add a similar level of explanation here ? 

Also, we do special handling for date expectedType. I am not sure we need to handle timestamp types too ?

Reply:
Added comments :)

So thing interesting is that we get a parsed ConstantExpression for `TIMESTAMP`/`TIMESTAMP_WITH_TIME_ZONE` but not `DATE`.

For example,
- `DATE '2014-01-31'`, we get a cast function on top of ConstantExpression of VarChar type
![image](https://user-images.githubusercontent.com/1202120/85001715-db943780-b108-11ea-800e-6c87938014a5.png)

- TIMESTAMP '2014-01-31 00:00:00 UTC', we directly get a ConstantExpression of LONG value.
![image](https://user-images.githubusercontent.com/1202120/85001621-bc95a580-b108-11ea-9127-e77b5e95b8b2.png)